python/cuml/common/array.py

```diff
@@ -349,19 +349,6 @@ def serialize(self) -> Tuple[dict, list]:
         frames = [CumlArray(f) for f in frames]
         return header, frames
 
-    @nvtx.annotate(message="common.CumlArray.copy", category="utils",
-                   domain="cuml_python")
-    def copy(self) -> Buffer:
```

Comment:
cuml Array will no longer inherit from cuDF Buffer very soon (see @wphicks PR [for CPU/GPU](https://github.com/rapidsai/cuml/pull/4908)) and it will very likely diverge more from it, so cuDF's Buffer changes don't need to be mirrored. What are the advantages of removing it? 

I would at least consider doing the change *after* PR 4908 goes in with all the changes that it is doing to cumlarray, which is still in fluctuation.

Reply:
> What are the advantages of removing it?

Removal of unused code. In its current state, calling `.copy()` raises an exception. 
It should probably also be removed in https://github.com/rapidsai/cuml/pull/4908: https://github.com/rapidsai/cuml/blob/e6c56a8ad32f841c40e3237a46189f8537aa221e/python/cuml/internals/array.py#L639-L649

Reply:
Looking back at why I added it, it was a provisional fix when cudf Buffer did some changes but ended up going a different route to fix things, and it is now wrong and not needed indeed, so fully agree, will merge!